<a href="https://colab.research.google.com/github/tomyhrdnsyh/Analisis-Sentimen-Twitter-Vaksin-Covid-19/blob/main/Training_%26_Testing.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd
import numpy as np

In [3]:
path = "/content/Tweet_Preprocessing.csv"
TWEET_DATA = pd.read_csv(path, usecols=["sentimen", "tweet_normalized"])
TWEET_DATA.columns = ["lable", "tweet"]
TWEET_DATA.lable = TWEET_DATA.lable.str.capitalize()
TWEET_DATA

lable                                              tweet
0    Negatif  ['nikah', 'tahun', 'initapi', 'wna', 'boleh', ...
1    Positif     ['semangat', 'jokowimempertahankan', 'negara']
2    Negatif  ['jokowi', 'tolong', 'psbb', 'selesaikan', 'se...
3    Positif                                          ['sehat']
4    Negatif  ['bagaimana', 'mau', 'berjuang', 'suami', 'sud...
..       ...                                                ...
995  Negatif  ['sibuk', 'vaksinbiar', 'menkes', 'tugasnyaleb...
996  Positif       ['mudahan', 'tenaga', 'vaksin', 'jt', 'wni']
997  Positif  ['terima', 'kasih', 'bapak', 'presiden', 'sema...
998  Positif  ['kaltim', 'samarinda', 'hadir', 'salam', 'seh...
999  Positif                                         ['vaksin']

[1000 rows x 2 columns]

# CONVERT STRING TO LIST

In [4]:
# convert list formated string to list
import ast

def convert_text_list(texts):
    texts = ast.literal_eval(texts)
    return [text for text in texts]

TWEET_DATA["tweet_list"] = TWEET_DATA["tweet"].apply(convert_text_list)
pd.DataFrame(TWEET_DATA[["tweet_list", "lable"]])

tweet_list    lable
0    [nikah, tahun, initapi, wna, boleh, masuk, ind...  Negatif
1             [semangat, jokowimempertahankan, negara]  Positif
2    [jokowi, tolong, psbb, selesaikan, secepatnya,...  Negatif
3                                              [sehat]  Positif
4    [bagaimana, mau, berjuang, suami, sudah, kerja...  Negatif
..                                                 ...      ...
995  [sibuk, vaksinbiar, menkes, tugasnyalebih, vis...  Negatif
996                 [mudahan, tenaga, vaksin, jt, wni]  Positif
997       [terima, kasih, bapak, presiden, semangatya]  Positif
998  [kaltim, samarinda, hadir, salam, sehat, semog...  Positif
999                                           [vaksin]  Positif

[1000 rows x 2 columns]

## Split Dataset

In [5]:
training_data = TWEET_DATA.sample(frac=0.8, random_state=50).reset_index()
training_data = training_data.drop(columns='index')
testing_data = TWEET_DATA.drop(training_data.index).reset_index()
testing_data = testing_data.drop(columns='index')

In [6]:
training_data

lable  ...                                         tweet_list
0    Positif  ...           [jokowi, masuk, daftar, vaksin, ditahap]
1    Positif  ...                                       [sakit, itu]
2    Negatif  ...                                            [sakit]
3    Negatif  ...                           [bhineka, tinggal, duka]
4    Positif  ...                                      [terimakasih]
..       ...  ...                                                ...
795  Positif  ...  [tunggu, reaksi, vaksin, dari, bapak, berhasil...
796  Positif  ...                        [menghadapi, wabah, corona]
797  Positif  ...  [disiplin, patuh, protokol, kesehatan, insha, ...
798  Positif  ...  [mudah, mudahan, vaksin, halal, lupa, ikhtiar,...
799  Positif  ...  [semangat, papresku, semoga, sehat, walafiat, ...

[800 rows x 3 columns]

In [7]:
testing_data

lable  ...                                         tweet_list
0    Positif  ...                     [semangat, jokowi, presidenku]
1    Negatif  ...  [lapor, masyarakat, seperti, kasih, vaksin, se...
2    Positif  ...                                 [semangat, jokowi]
3    Negatif  ...  [jokowibln, februari, bansos, uang, tunai, bel...
4    Positif  ...  [semoga, indonesia, mengatasi, pandemi, allah,...
..       ...  ...                                                ...
195  Negatif  ...  [sibuk, vaksinbiar, menkes, tugasnyalebih, vis...
196  Positif  ...                 [mudahan, tenaga, vaksin, jt, wni]
197  Positif  ...       [terima, kasih, bapak, presiden, semangatya]
198  Positif  ...  [kaltim, samarinda, hadir, salam, sehat, semog...
199  Positif  ...                                           [vaksin]

[200 rows x 3 columns]

## Split Pos Neg Dataset

In [8]:
training_data_pos = training_data[['lable', 'tweet_list']]
training_data_pos = training_data_pos[training_data_pos.lable == 'Positif'].reset_index()
training_data_pos = training_data_pos.drop(columns='index')
training_data_pos

lable                                         tweet_list
0    Positif           [jokowi, masuk, daftar, vaksin, ditahap]
1    Positif                                       [sakit, itu]
2    Positif                                      [terimakasih]
3    Positif  [mantap, bapak, presidenkuslm, doa, tuhan, mah...
4    Positif  [assalamualaikumsmga, bapak, presiden, joko, w...
..       ...                                                ...
408  Positif  [tunggu, reaksi, vaksin, dari, bapak, berhasil...
409  Positif                        [menghadapi, wabah, corona]
410  Positif  [disiplin, patuh, protokol, kesehatan, insha, ...
411  Positif  [mudah, mudahan, vaksin, halal, lupa, ikhtiar,...
412  Positif  [semangat, papresku, semoga, sehat, walafiat, ...

[413 rows x 2 columns]

In [9]:
training_data_neg = training_data[['lable', 'tweet_list']]
training_data_neg = training_data_neg[training_data_neg.lable == 'Negatif'].reset_index()
training_data_neg = training_data_neg.drop(columns='index')
training_data_neg

lable                                         tweet_list
0    Negatif                                            [sakit]
1    Negatif                           [bhineka, tinggal, duka]
2    Negatif  [terhormat, bali, terpuruk, ekonominya, ppkm, ...
3    Negatif              [kampung, petani, sush, nyari, pupuk]
4    Negatif                               [aturan, bermanfaat]
..       ...                                                ...
382  Negatif                   [tipu, suntik, jarumnya, pendek]
383  Negatif                [cari, duit, susahrakyat, sengsara]
384  Negatif  [vaksin, psbb, berjalan, lapangan, pekerjaan, ...
385  Negatif  [perkenalkan, nama, putri, lestari, hambali, s...
386  Negatif                                          [sekolah]

[387 rows x 2 columns]

**Menghitung Bobot TF**

In [17]:
def calc_TF(document):
    # Counts the number of times the word appears in review
    TF_dict = {}
    for term in document:
        if term in TF_dict:
            TF_dict[term] += 1
        else:
            TF_dict[term] = 1
    for term in TF_dict:
        TF_dict[term] = TF_dict[term] / len(document)
    return TF_dict

training_data_pos["TF_dict"] = training_data_pos['tweet_list'].apply(calc_TF)
training_data_neg["TF_dict"] = training_data_neg['tweet_list'].apply(calc_TF)
training_data["TF_dict"] = training_data['tweet_list'].apply(calc_TF)
training_data_pos

lable  ...                                            TF_dict
0    Positif  ...  {'jokowi': 0.2, 'masuk': 0.2, 'daftar': 0.2, '...
1    Positif  ...                         {'sakit': 0.5, 'itu': 0.5}
2    Positif  ...                               {'terimakasih': 1.0}
3    Positif  ...  {'mantap': 0.07692307692307693, 'bapak': 0.076...
4    Positif  ...  {'assalamualaikumsmga': 0.09090909090909091, '...
..       ...  ...                                                ...
408  Positif  ...  {'tunggu': 0.1111111111111111, 'reaksi': 0.111...
409  Positif  ...  {'menghadapi': 0.3333333333333333, 'wabah': 0....
410  Positif  ...  {'disiplin': 0.07142857142857142, 'patuh': 0.0...
411  Positif  ...  {'mudah': 0.1111111111111111, 'mudahan': 0.111...
412  Positif  ...  {'semangat': 0.14285714285714285, 'papresku': ...

[413 rows x 3 columns]

In [11]:
def call_tf(query):
    call_tf = {}
    index = query.index
    for i in index:
        for term in query[i]:
            if term in call_tf:
                call_tf[term] += query[i][term]
            else:
                call_tf[term] = query[i][term]
    return call_tf

In [15]:
TF_Pos = call_tf(training_data_pos["TF_dict"])
df_TF_Pos = pd.DataFrame(TF_Pos.items(), columns=["Istilah", "TF-Positif"])

TF_Neg = call_tf(training_data_neg["TF_dict"])
df_TF_Neg = pd.DataFrame(TF_Neg.items(), columns=["Istilah", "TF-Negatif"])

In [16]:
df_TF_Pos

Istilah  TF-Positif
0       jokowi   15.100185
1        masuk    1.805357
2       daftar    0.200000
3       vaksin   12.397893
4      ditahap    0.200000
...        ...         ...
1058   ikhtiar    0.111111
1059  mendekat    0.111111
1060      sang    0.111111
1061    khalik    0.111111
1062  papresku    0.142857

[1063 rows x 2 columns]

**Menghitung Bobot DF**

In [13]:
def calc_DF(tfDict):
    count_DF = {}
    # Run through each document's tf dictionary and increment countDict's (term, doc) pair
    for document in tfDict:
        for term in document:
            if term in count_DF:
                count_DF[term] += 1
            else:
                count_DF[term] = 1
    return count_DF

In [25]:
DF = calc_DF(training_data["TF_dict"])

**Menghitung Bobot IDF**

In [26]:
def calc_IDF(__n_document, __DF):
    # print("Menghitung IDF\n")
    IDF_Dict = {}
    d = {}
    for term in __DF:
        IDF_Dict[term] = np.log10(__n_document / (__DF[term]))
    return IDF_Dict

In [27]:
# Stores the idf dictionary
n_document = len(training_data)
IDF = calc_IDF(n_document, DF)

In [28]:
def rumus_IDF(__n_document, __DF):
    IDF_Dict = {}
    d = {}
    df = {}
    rumus = {}
    for term in __DF:
        IDF_Dict[term] = np.log10(__n_document / (__DF[term]))
        d[term] = __n_document
        df[term] = __DF[term]
        rumus[term] = 'log({}/{})'.format(round(d[term], 6), round(df[term], 4)) 
    IDF_Dict = pd.DataFrame({ 'D':pd.Series(d), 'DF':pd.Series(df), 'Rumus IDF = log(D/DF)':pd.Series(rumus), 'IDF':pd.Series(IDF_Dict)}).reset_index()
    return IDF_Dict

In [29]:
df_IDF = rumus_IDF(n_document, DF)
df_IDF = df_IDF.rename(columns={'index': 'Istilah'})
df_IDF

Istilah    D  DF Rumus IDF = log(D/DF)       IDF
0       jokowi  800  84           log(800/84)  0.978811
1        masuk  800  28           log(800/28)  1.455932
2       daftar  800   2            log(800/2)  2.602060
3       vaksin  800  89           log(800/89)  0.953700
4      ditahap  800   1            log(800/1)  2.903090
...        ...  ...  ..                   ...       ...
3086   alamiin  800   1            log(800/1)  2.903090
3087   ikhtiar  800   1            log(800/1)  2.903090
3088  mendekat  800   1            log(800/1)  2.903090
3089    khalik  800   1            log(800/1)  2.903090
3090  papresku  800   1            log(800/1)  2.903090

[3091 rows x 5 columns]

**Total IDF**

In [30]:
print('Total IDF = ', df_IDF['IDF'].sum())

Total IDF =  8510.16355844541


**Menghitung Bobot TFIDF**

In [31]:
#calc TF-IDF
def calc_TF_IDF(TF):
    TF_IDF_Dict = {}
    #For each word in the review, we multiply its tf and its idf.
    for key in TF:
        TF_IDF_Dict[key] = TF[key] * IDF[key]
    return TF_IDF_Dict

TFIDF Positif Negatif

In [33]:
training_data_pos["TF-IDF_dict"] = training_data_pos["TF_dict"].apply(calc_TF_IDF)
training_data_pos

lable  ...                                        TF-IDF_dict
0    Positif  ...  {'jokowi': 0.1957621401860124, 'masuk': 0.2911...
1    Positif  ...  {'sakit': 0.8363205328068348, 'itu': 1.1505149...
2    Positif  ...                {'terimakasih': 1.8616973018337186}
3    Positif  ...  {'mantap': 0.12160543786600186, 'bapak': 0.103...
4    Positif  ...  {'assalamualaikumsmga': 0.26391727154472217, '...
..       ...  ...                                                ...
408  Positif  ...  {'tunggu': 0.26955208136358677, 'reaksi': 0.32...
409  Positif  ...  {'menghadapi': 0.8673533304426542, 'wabah': 0....
410  Positif  ...  {'disiplin': 0.1643592854045701, 'patuh': 0.18...
411  Positif  ...  {'mudah': 0.289117776814218, 'mudahan': 0.2449...
412  Positif  ...  {'semangat': 0.18283438522772044, 'papresku': ...

[413 rows x 4 columns]

In [34]:
training_data_neg["TF-IDF_dict"] = training_data_neg["TF_dict"].apply(calc_TF_IDF)
training_data_neg

lable  ...                                        TF-IDF_dict
0    Negatif  ...                      {'sakit': 1.6726410656136697}
1    Negatif  ...  {'bhineka': 0.9676966623306479, 'tinggal': 0.7...
2    Negatif  ...  {'terhormat': 0.030303030303030304, 'bali': 0....
3    Negatif  ...  {'kampung': 0.48519374645445623, 'petani': 0.5...
4    Negatif  ...  {'aturan': 1.1020599913279625, 'bermanfaat': 1...
..       ...  ...                                                ...
382  Negatif  ...  {'tipu': 0.7257724967479859, 'suntik': 0.43174...
383  Negatif  ...  {'cari': 0.5, 'duit': 0.5510299956639813, 'sus...
384  Negatif  ...  {'vaksin': 0.012226922824961933, 'psbb': 0.042...
385  Negatif  ...  {'perkenalkan': 0.060513023054138665, 'nama': ...
386  Negatif  ...                     {'sekolah': 1.411728293157671}

[387 rows x 4 columns]

# TF-IDF Dataframe

In [35]:
def rumus_TF_IDF(__tf, __idf):
  __TF = {}
  __IDF =  {}
  rumus_TF_IDF = {}
  for key in __tf:
    __TF[key] = __tf[key]
    __IDF[key] = __idf[key]
    rumus_TF_IDF[key] = '{} * {}'.format(round(__tf[key], 6), round(__idf[key], 4)) 

  rumus_TF_IDF = pd.DataFrame({'TF':pd.Series(__TF), 'IDF':pd.Series(__IDF), 'Rumus':pd.Series(rumus_TF_IDF)}).reset_index()
  return rumus_TF_IDF 

In [36]:
def fix_tf_idf(query):
    tf_idf_ = {}
    kemunculan = {}
    index = query.index
    for i in index:
        for term in query["TF-IDF_dict"][i]:
          if term in tf_idf_:
                tf_idf_[term] += query["TF-IDF_dict"][i][term]
                kemunculan[term] += 1
          else:
                tf_idf_[term] = query["TF-IDF_dict"][i][term]
                kemunculan[term] = 1
    tf_idf_ = pd.DataFrame({'Kemunculan':pd.Series(kemunculan), 'TF-IDF':pd.Series(tf_idf_)}).reset_index()
    return tf_idf_

**TF-IDF POSITIF**

In [37]:
fix_tfidf_pos = fix_tf_idf(training_data_pos)
fix_tfidf_pos.columns = ['Istilah Positif', 'Kemunculan', 'TF-IDF']
rumus = rumus_TF_IDF(TF_Pos, IDF)
fix_tfidf_pos.insert(2, 'TF', rumus['TF'])
fix_tfidf_pos.insert(3, 'IDF', rumus['IDF'])
fix_tfidf_pos.insert(4, 'TF-IDF=TF*IDF', rumus['Rumus'])
fix_tfidf_pos

Istilah Positif  Kemunculan  ...       TF-IDF=TF*IDF     TF-IDF
0             jokowi          60  ...  15.100185 * 0.9788  14.780223
1              masuk           9  ...   1.805357 * 1.4559   2.628477
2             daftar           1  ...        0.2 * 2.6021   0.520412
3             vaksin          47  ...  12.397893 * 0.9537  11.823871
4            ditahap           1  ...        0.2 * 2.9031   0.580618
...              ...         ...  ...                 ...        ...
1058         ikhtiar           1  ...   0.111111 * 2.9031   0.322566
1059        mendekat           1  ...   0.111111 * 2.9031   0.322566
1060            sang           1  ...   0.111111 * 2.6021   0.289118
1061          khalik           1  ...   0.111111 * 2.9031   0.322566
1062        papresku           1  ...   0.142857 * 2.9031   0.414727

[1063 rows x 6 columns]

**TFIDF NEGATIF**

In [38]:
fix_tfidf_neg = fix_tf_idf(training_data_neg)
fix_tfidf_neg.columns = ['Istilah Negatif', 'Kemunculan', 'TF-IDF']
rumus = rumus_TF_IDF(TF_Neg, IDF)
fix_tfidf_neg.insert(2, 'TF', rumus['TF'])
fix_tfidf_neg.insert(3, 'IDF', rumus['IDF'])
fix_tfidf_neg.insert(4, 'TF-IDF=TF*IDF', rumus['Rumus'])

fix_tfidf_neg

Istilah Negatif  Kemunculan  ...      TF-IDF=TF*IDF    TF-IDF
0              sakit          11  ...   1.67552 * 1.6726  2.802544
1            bhineka           1  ...  0.333333 * 2.9031  0.967697
2            tinggal           5  ...  0.668943 * 2.1249  1.421462
3               duka           1  ...  0.333333 * 2.9031  0.967697
4          terhormat           7  ...     0.472125 * 2.0  0.944250
...              ...         ...  ...                ...       ...
2425       kabupaten           1  ...  0.023256 * 2.6021  0.060513
2426     berkomentar           1  ...   0.023256 * 2.426  0.056418
2427       postingan           1  ...  0.023256 * 2.9031  0.067514
2428      terkenanya           1  ...  0.023256 * 2.9031  0.067514
2429         jaminan           1  ...  0.023256 * 2.9031  0.067514

[2430 rows x 6 columns]

Menghitung Total TFIDF

In [39]:
print('Total TF-IDF Pos', fix_tfidf_pos['TF-IDF'].sum())

Total TF-IDF Pos 784.9308385990958


MENGHITUNG TFIDF NEGATIF

In [40]:
print('Total TF-IDF Neg', fix_tfidf_neg['TF-IDF'].sum())

Total TF-IDF Neg 861.9607794546077


TOTAL TFIDF POSITIF DAN NEGATIF

In [41]:
total_tfidf_full = fix_tfidf_pos['TF-IDF'].sum()+fix_tfidf_neg['TF-IDF'].sum()
print('Total TF-IDF Keseluruhan = ', total_tfidf_full)

Total TF-IDF Keseluruhan =  1646.8916180537035


# Testing Data

In [42]:
testing_data

lable  ...                                         tweet_list
0    Positif  ...                     [semangat, jokowi, presidenku]
1    Negatif  ...  [lapor, masyarakat, seperti, kasih, vaksin, se...
2    Positif  ...                                 [semangat, jokowi]
3    Negatif  ...  [jokowibln, februari, bansos, uang, tunai, bel...
4    Positif  ...  [semoga, indonesia, mengatasi, pandemi, allah,...
..       ...  ...                                                ...
195  Negatif  ...  [sibuk, vaksinbiar, menkes, tugasnyalebih, vis...
196  Positif  ...                 [mudahan, tenaga, vaksin, jt, wni]
197  Positif  ...       [terima, kasih, bapak, presiden, semangatya]
198  Positif  ...  [kaltim, samarinda, hadir, salam, sehat, semog...
199  Positif  ...                                           [vaksin]

[200 rows x 3 columns]

In [43]:
def calc_probability(uji, kelas, idf):
    calc_prob = {}
    for term in uji:
        if term in kelas.index:
          calc_prob[term] = (kelas['TF-IDF'][term]+1)/(kelas['TF-IDF'].sum()+idf['IDF'].sum())
        else:
          calc_prob[term] = (0+1)/(kelas['TF-IDF'].sum()+idf['IDF'].sum())
    return calc_prob

In [44]:
kelas_pos = fix_tfidf_pos.set_index('Istilah Positif')
kelas_neg = fix_tfidf_neg.set_index('Istilah Negatif')

In [45]:
testing_data['Cond Probability Positif'] = testing_data['tweet_list'].apply(calc_probability, kelas=(kelas_pos), idf=(df_IDF))
testing_data['Cond Probability Negatif'] = testing_data['tweet_list'].apply(calc_probability, kelas=(kelas_neg), idf=(df_IDF))
testing_data

lable  ...                           Cond Probability Negatif
0    Positif  ...  {'semangat': 0.0001476083628143587, 'jokowi': ...
1    Negatif  ...  {'lapor': 0.00014011881165207807, 'masyarakat'...
2    Positif  ...  {'semangat': 0.0001476083628143587, 'jokowi': ...
3    Negatif  ...  {'jokowibln': 0.00013251255753836231, 'februar...
4    Positif  ...  {'semoga': 0.00019436075686825598, 'indonesia'...
..       ...  ...                                                ...
195  Negatif  ...  {'sibuk': 0.00023163659783284686, 'vaksinbiar'...
196  Positif  ...  {'mudahan': 0.00010669939535011192, 'tenaga': ...
197  Positif  ...  {'terima': 0.00012492308829515364, 'kasih': 0....
198  Positif  ...  {'kaltim': 0.0001414041738175383, 'samarinda':...
199  Positif  ...                  {'vaksin': 0.0007347271499300473}

[200 rows x 5 columns]

In [46]:
def Posterior_Probability(conditional_prob, prior_probability):
    posterior_prob = 1
    for key in conditional_prob:
        posterior_prob = posterior_prob*conditional_prob[key]
    posterior_prob = posterior_prob*prior_probability
    return posterior_prob

In [47]:
total_train = len(training_data_pos)+len(training_data_neg)
prior_probability_pos = len(training_data_pos)/total_train
prior_probability_neg = len(training_data_neg)/total_train

testing_data['Post_Prob_Pos'] = testing_data['Cond Probability Positif'].apply(Posterior_Probability, prior_probability=(prior_probability_pos))
testing_data['Post_Prob_Neg'] = testing_data['Cond Probability Negatif'].apply(Posterior_Probability, prior_probability=(prior_probability_neg))

In [48]:
testing_data

lable  ... Post_Prob_Neg
0    Positif  ...  2.429857e-12
1    Negatif  ...  6.589020e-34
2    Positif  ...  2.035217e-08
3    Negatif  ...  1.372629e-46
4    Positif  ...  2.495354e-23
..       ...  ...           ...
195  Negatif  ...  1.224943e-76
196  Positif  ...  9.921209e-20
197  Positif  ...  1.970171e-19
198  Positif  ...  5.533151e-28
199  Positif  ...  3.554243e-04

[200 rows x 7 columns]

In [49]:
testing_data['Prediksi'] = np.where(testing_data['Post_Prob_Pos'] >= testing_data['Post_Prob_Neg'], 'Positif', 'Negatif')
testing_data.lable = testing_data.lable.str.capitalize()
testing_data

lable  ... Prediksi
0    Positif  ...  Positif
1    Negatif  ...  Negatif
2    Positif  ...  Positif
3    Negatif  ...  Negatif
4    Positif  ...  Positif
..       ...  ...      ...
195  Negatif  ...  Negatif
196  Positif  ...  Positif
197  Positif  ...  Positif
198  Positif  ...  Positif
199  Positif  ...  Positif

[200 rows x 8 columns]

In [50]:
testing_data['Kesesuaian'] = np.where(testing_data['Prediksi'] == testing_data['lable'], 'Sesuai', 'Tidak Sesuai')
testing_data

lable  ... Kesesuaian
0    Positif  ...     Sesuai
1    Negatif  ...     Sesuai
2    Positif  ...     Sesuai
3    Negatif  ...     Sesuai
4    Positif  ...     Sesuai
..       ...  ...        ...
195  Negatif  ...     Sesuai
196  Positif  ...     Sesuai
197  Positif  ...     Sesuai
198  Positif  ...     Sesuai
199  Positif  ...     Sesuai

[200 rows x 9 columns]

In [51]:
sesuai = len(testing_data[testing_data.Kesesuaian == 'Sesuai'])
total = len(testing_data)
akurasi = (sesuai/total)*100

In [52]:
print('Akurasi = {} %'.format(akurasi))

Akurasi = 92.5 %


## Kesimpulan

Analisis sentimen menggunakan naive bayes ini diuji dengan
menggunakan 800 data latih dan 200 data uji. Kata unik yang dihasilkan dari 1000
komentar adalah 3091 kata unik, sedangkan komentar bersentimen positif memiliki
1063 kata unik dan komentar bersentimen negatif memiliki 2430 kata unik. Akurasi
yang dihasilkan menggunakan 800 data latih dan 200 data uji adalah 92.5%. Artinya sistem analisis sentimen mampu menganalisis 185 komentar dengan benar dari 200 komentar yang digunakan